src/playermat/Playmat.ttslua

```diff
@@ -546,17 +546,30 @@ function replenishTokens(card, count, replenish)
   -- don't continue for cards on your deck (Norman) or in your discard pile
   if self.positionToLocal(cardPos).x < -1 then return end
 
+  -- resource token states
+  local stateTable = {
+    ["resource"] = 1,
+    ["ammo"] = 2,
+    ["charge"] = 3,
+    ["secret"] = 4,
+    ["supply"] = 5
+  }
+  local resourceState = 1
+
   -- get current amount of resource tokens on the card
   local search = searchArea(cardPos, { 2.5, 0.5, 3.5 })
   local clickableResourceCounter = nil
   local foundTokens = 0
 
   for _, obj in ipairs(search) do
     local obj = obj.hit_object
-    if obj.getCustomObject().image == "http://cloud-3.steamusercontent.com/ugc/1758068501357192910/11DDDC7EF621320962FDCF3AE3211D5EDC3D1573/" then
+    local memo = obj.getMemo()
+
+    if (stateTable[memo] or 0) > 0 then
```

Comment:
This is going to be pretty inconsistent if a card ends up with different token types, since it'll just take the last one found.  I think it'll be better to pass the type in, and spawn that explicitly?